src/MPAGSCipher/CaesarCipher.cpp

```diff
@@ -0,0 +1,90 @@
+#include "CaesarCipher.hpp"
+#include "CipherMode.hpp"
+
+#include <string>
+#include <iostream>
+
+// class CaesarCipher {
+//     public:
+//     explicit CaesarCipher(const std::size_t& key);
+
+//     explicit CaesarCipher(const std::string& key);
+
+//     std::string applyCipher (const std::string& inputText, const CipherMode encrypt) const;
+
+//     private:
+
+//     std::size_t key_ ;
+//     const std::vector<char> alphabet_;
+
+
+// };
+
+CaesarCipher::CaesarCipher(const std::size_t& key)
+:key_{0},alphabet_{'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'}
+{
+    key_ = key;
+}
+
+CaesarCipher::CaesarCipher(const std::string& key)
+:key_{0}, alphabet_{'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'}
+{
+    if (!key.empty()) {
+    // Before doing the conversion we should check that the string contains a
+    // valid positive integer.
+    // Here we do that by looping through each character and checking that it
+    // is a digit. What is rather hard to check is whether the number is too
+    // large to be represented by an unsigned long, so we've omitted that for
+    // the time being.
+    // (Since the conversion function std::stoul will throw an exception if the
+    // string does not represent a valid unsigned long, we could check for and
+    // handled that instead but we only cover exceptions very briefly on the
+    // final day of this course - they are a very complex area of C++ that
+    // could take an entire course on their own!)
+    for (const auto& elem : key) {
+        if (!std::isdigit(elem)) {
+            std::cerr
+                << "[error] cipher key must be an unsigned long integer for Caesar cipher,\n"
+                << "        the supplied key (" << key
+                << ") could not be successfully converted" << std::endl;
+        }
```

Comment:
If you encounter a problem you need to bail out before you attempt to convert the key, otherwise you'll get an exception thrown from `std::stoul`.  So just return here.
```suggestion
        if (!std::isdigit(elem)) {
            std::cerr
                << "[error] cipher key must be an unsigned long integer for Caesar cipher,\n"
                << "        the supplied key (" << key
                << ") could not be successfully converted" << std::endl;
            return;
        }
```